### DISCUSSION TITLE
Delete trigger and cascade delete

### DISCUSSION THREAD

**Discussion by hamanuha** [supabase/supabase#12538] [Questions] [answered]
Hi there,

I am using a before delete trigger to prevent the deletion of the last workspace for any user.
But if I want to delete the whole user with all spaces, it's not possible due to the trigger.
What would be the best way to handle this? Move the check to the api server?

```sql
create function public.handle_delete_space()
returns trigger
language plpgsql
as $$
  begin
    -- Prevent deletion of last space
    if (select count(id) from public.spaces where owner_id = old.owner_id) = 1 then
      raise exception 'The last space cannot be deleted';
    end if;

    return old;
  end;
$$;

create trigger on_space_delete
  before delete on public.spaces
  for each row execute function public.handle_delete_space();
```

**Comment by GaryAustin1** [comment, marked as answer]
Possibly this from a quick Google search:
https://stackoverflow.com/questions/24285658/how-to-check-if-deletion-is-caused-by-cascade-in-postgresql-trigger

Seems like more answers out there as this is standard Postgres stuff.

**Reply by hamanuha** [reply]
Thanks, did not found this. Only some other stuff that did not work.
